magnum_capi_helm/driver.py

```diff
@@ -578,6 +596,7 @@ def _update_helm_release(self, context, cluster, nodegroups=None):
                     ),
                 },
             },
+            "osDistro": os_distro,
```

Comment:
oh... are we looking for flatcar here, rather than kubeadm-ignition?

@mkjpryor can you confirm what the helm charts are wanting here for ignition vs cloud-init?

Reply:
So the charts only specifically look for `flatcar`, and do the ubuntu stuff for anything else. We were just discussing the driver selection in the IRC meeting and it looks like we will be able to move back to `ubuntu` for the `os_distro` (which is my preference as it is true), and use an additional image property for tie-breaks between drivers.

Reply:
So my problem with that is that it intentionally breaks nova and conflicts with the out of tree drivers.

Reply:
The idea is that we will have an additional image property that just specifies the driver that the image is for. If present, this will be used to select the driver, falling back to `(coe, os, type)` as now.

Jake's spec is here for comment: https://review.opendev.org/c/openstack/magnum-specs/+/900410/1/specs/caracal/improve-driver-discovery.rst

Reply:
Yeah, the extra driver property makes sense.

I had assumed that cloudinit vs ignition is enough to know what to do, but you are telling me that isn't the case because of other config we are injecting, relating to things like the directory for containerd config. So it does seem pragmatic to go for flatcar vs ubuntu here.

Reply:
Thanks for the conversation here, I also wasn't sure why the other PR went with `flatcar` vs the seemingly simpler `ignition`.